lib/instrumentation/modules/memcached.js

```diff
@@ -0,0 +1,47 @@
+'use strict'
+
+var semver = require('semver')
+
+var shimmer = require('../shimmer')
+
+module.exports = function (memcached, agent, { version, enabled }) {
+  if (!enabled) {
+    return memcached
+  }
+  if (!semver.satisfies(version, '>=2.2.0')) {
+    agent.logger.debug('Memcached version %s not supported - aborting...', version)
+    return memcached
+  }
+  agent.logger.debug('shimming memcached.prototype.command')
+  shimmer.wrap(memcached.prototype, 'command', wrapCommand)
+  return memcached
+
+  // Wrap the generic command that is used to build touch, get, gets etc
+  function wrapCommand (original) {
+    return function wrappedCommand () {
+      if (typeof arguments[0] === 'function') {
+        var query = arguments[0]()
+        // If the callback is not a function the user doesn't care about result
+        if (query && typeof query.callback === 'function') {
+          var span = agent.startSpan(`memcached.${query.type}`, `db.memcached.${query.type}`)
+          span.setDbContext({ statement: `${query.type} ${query.key}`, type: 'memcached' })
```

Comment:
Missed this one the last time around:

The return value of `startSpan()` can be `null` depending on how the agent is operating, so we need to add a guard on this line to be sure we don't get an error.

But since we already have one on the line after, I suggest we just wrap both this and the next line in an if-block, eg:

```js
if (span) {
  span.setDbContext({ statement: `${query.type} ${query.key}`, type: 'memcached' })
  var id = span.id
}
```